js/components/graph/Graph.js

```diff
@@ -93,19 +93,7 @@ export default class Graph extends React.Component {
 
   getGraph = graphName => {
     if (graphName === undefined) {
-      const params = new URL(document.location).searchParams;
-      const urlSpecifiedGraph = params.get("dept");
-
-      // HACK: Temporary workaround for giving the statistics department a
-      // link to our graph.
-      // Should be replaced with a more general solution.
-      if (urlSpecifiedGraph === "sta") {
-        graphName = "Statistics";
-      } else if (urlSpecifiedGraph !== null) {
-        graphName = "Computer Science";
-      } else {
-        graphName = localStorage.getItem("active-graph") || "Computer Science";
-      }
+      graphName = this.props.getLocalGraph();
```

Comment:
Rather than passing `getLocalGraph` down as a prop, pass the `graphName` itself down as a prop. You should be able to make `getGraph` a function that takes no arguments, and always retrieves `this.props.graphName`. Call this function in a `componentDidUpdate` when the prop changes.

(Also, when you're doing this, move the `replace` call on line 99 up to the `Container` class.

Reply:
Had to call the function in `componentWillUpdate` because changes did not propogate.